<?xml version="1.0"?>
<launch>
    <arg name="rviz_display" default="true"/>
    <group if = "$(arg rviz_display)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find dmvc_tracker3d)/launch/rviz_config/dynamic_obstacle.rviz" output="log"/>
    </group>
    <node pkg="dmvc_tracker3d" type="dmvc_simulator3d" name="dmvc_simulator3d" output="screen">
        <rosparam command="load" file="$(find dmvc_tracker3d)/param/dynamic_obstacle/101object.yaml"/>
        <param name="object_history_file_name" value="$(find dmvc_tracker3d)/world/dynamic_obstacle/dynamic_object_history101.csv"/>
        <param name="initial_state_file_name" value="$(find dmvc_tracker3d)/world/dynamic_obstacle/initial_trajectory_101object.txt"/>
    </node>

    <node pkg="dmvc_tracker3d" type="dmvc_predictor3d" name="dmvc_predictor3d" output="screen">
        <rosparam command="load" file="$(find dmvc_tracker3d)/param/dynamic_obstacle/prediction_visualizer_100.yaml"/>
        <rosparam command="load" file="$(find dmvc_tracker3d)/param/dynamic_obstacle/predictor_100.yaml"/>
    </node>

    <node pkg="dmvc_tracker3d" type="dmvc_tracker3d" name="dmvc_tracker3d" output="screen">
        <rosparam command="load" file="$(find dmvc_tracker3d)/param/dynamic_obstacle/planning_visualizer_100.yaml"/>
        <rosparam command="load" file="$(find dmvc_tracker3d)/param/dynamic_obstacle/planner_100.yaml"/>
    </node>
</launch>